metric_learn/_util.py

```diff
@@ -324,7 +326,32 @@ def check_collapsed_pairs(pairs):
                        "in total.".format(num_ident, pairs.shape[0]))
 
 
-def transformer_from_metric(metric):
+def _check_sdp_from_eigen(w, tol=None):
+  """Checks if some of the eigenvalues given are negative, up to a tolerance
+  level, with a default value of the tolerance depending on the eigenvalues.
+
+  Parameters
+  ----------
+  w : array-like, shape=(n_eigenvalues,)
+    Eigenvalues to check for non semidefinite positiveness.
+
+  tol : float, optional
+    Negative eigenvalues above - tol are considered zero. If
+    tol is None, and w are `metric`'s eigenvalues, and eps is the
+    epsilon value for datatype of w, then tol is set to w.max() * len(w) * eps.
+
+  See Also
+  --------
+  np.linalg.matrix_rank for more details on the choice of tolerance (the same
+    strategy is applied here)
+  """
+  if tol is None:
+    tol = w.max() * len(w) * np.finfo(w.dtype).eps
+  if any(w[w < 0] < - tol):

```

Comment:
`w<0` not needed

Reply:
although it actually prevents that anything is done in case the provided `tol` is negative (which is not supposed to be the case according to the docstring but is not checked)

Reply:
That's right, I'll let it be w < - tol and add a quick test to check that tol is positive